.circleci/fail_fast.sh

```diff
@@ -16,17 +16,52 @@
 
 # This script early-exits CircleCI PR builds when one of its jobs fails.
 # Reference: https://support.circleci.com/hc/en-us/articles/360052058811-Exit-build-early-if-any-test-fails
+# In case of the `main` or a release branch, it emails the build/release-on-duty
+# team using Twilio SendGrid: https://sendgrid.com/
 
 set -e
 
 RED() { echo -e "\n\033[0;31m$1\033[0m"; }
 YELLOW() { echo -e "\n\033[0;33m$1\033[0m"; }
 
+send_email() {
+  curl --request POST \
+    --url https://api.sendgrid.com/v3/mail/send \
+    --header "Authorization: Bearer ${SENDGRID_API_KEY}" \
+    --header "Content-Type: application/json" \
+    --data '{
+        "from": {
+          "email": "bot+noreply@amp.dev"
+        },
+        "template_id": "d-5a6b574506534ab3aad1da13a78cdeb4",
+        "personalizations": [
+          {
+            "to": [
+              {
+                "email": "'"${1}"'",
+                "name": "'"${2}"'"
+              }
+            ],
+            "dynamic_template_data": {
+              "branch": "'"${CIRCLE_BRANCH}"'",
+              "name": "'"${2}"'",
+              "build_url": "'"${CIRCLE_BUILD_URL}"'"
+            }
+          }
+        ]
+      }'
+}
+
 # For push builds, continue in spite of failures so that other jobs like
 # bundle-size and visual-diff can establish their baselines for this commit.
 # Without this, our custom bots will not be able to function correctly.
-if [[ "$CIRCLE_BRANCH" == "main" || "$CIRCLE_BRANCH" =~ ^amp-release-* ]]; then
-  echo $(YELLOW "Not canceling build in spite of failures because $CIRCLE_BRANCH is not a PR branch.")
```

Comment:
This logging message and the comment on lines 25-27 are related to each other, and are meant to indicate why fast-fail isn't being applied to certain push builds.

Can we reinstate this message in both cases where we're not fast-failing?

Reply:
Done